designs/2019-worker-threads/README.md

```diff
@@ -22,27 +22,28 @@ This RFC has two steps.
 
 - Adds `--concurrency` CLI option to specify the number of worker threads. Defaults to `1`.
 - Adds `concurrency` constructor option to `ESLint` class to specify the number of worker threads. Defaults to `1`.
+- Adds `disallowWorkerThreads` option to plugins. Defaults to `false`.
 - Changes the behavior of `ESLint#executeOnFiles(patterns)` method.
 
 **The step 2 (semver-major):**
 
-- Changes the default of the `--concurrency` CLI option to `"auto"`.
+- Changes the default of the `--concurrency` CLI option to `auto`.
```

Comment:
Not sure why this is needed? Why just not use number of cpus to determine concurrency? I think we did some testing and determine that ESLint is CPU bound, not IO bound, so increasing number of threads beyond number of CPUs should provide none, or almost none improvements in speed. And in some cases, if the number of threads is too large, you will actually start seeing slowdown, due to the way CPUs are handling thread switching.

Reply:
Because creating workers has overhead.

Reply:
> I think we did some testing and determine that ESLint is CPU bound, not IO bound, so increasing number of threads beyond number of CPUs should provide none, or almost none improvements in speed.

This RFC creates threads the number of CPUs *at most* if `auto` is present. I guess that you are misreading something.